File: solarloop-ai/frontend/src/types/analysis.ts
export interface PanelZone {
  id: number;
  inspection_id: number;
  zone_code: string;
  row_index: number;
  col_index: number;
  x: number;
  y: number;
  width: number;
  height: number;
  soiling_score: number;
  shading_score: number;
  thermal_score: number;
  damage_score: number;
  generation_loss_score: number;
  priority_score: number;
  status_label: 'NORMAL' | 'SOILING' | 'SHADING' | 'THERMAL_ANOMALY' | 'PHYSICAL_DAMAGE_SUSPECTED' | 'GENERATION_LOSS_SUSPECTED';
  recommendation_label: 'NORMAL_MONITORING' | 'CLEANING_PRIORITY' | 'INSPECTION_REQUIRED' | 'WAIT_FOR_RAIN' | 'REPAIR_REVIEW';
  explanation?: string;
}

export interface MaintenanceAction {
  id: number;
  inspection_id: number;
  zone_code: string;
  action_type: 'NORMAL_MONITORING' | 'CLEANING_PRIORITY' | 'INSPECTION_REQUIRED' | 'WAIT_FOR_RAIN' | 'REPAIR_REVIEW';
  priority_rank: number;
  expected_recovery_kwh?: number;
  reason: string;
  created_at: string;
}

export interface AnalysisSummary {
  total_zones: number;
  normal_count: number;
  cleaning_priority_count: number;
  inspection_required_count: number;
  wait_for_rain_count: number;
  repair_review_count: number;
  average_priority_score: number;
}

File: solarloop-ai/backend/app/schemas/analysis.py
from pydantic import BaseModel, ConfigDict
from typing import List, Optional
from app.schemas.maintenance_action import MaintenanceAction

class PanelZoneBase(BaseModel):
    zone_code: str
    row_index: int
    col_index: int
    x: float
    y: float
    width: float
    height: float
    soiling_score: float
    shading_score: float
    thermal_score: float
    damage_score: float
    generation_loss_score: float
    priority_score: float
    status_label: str
    recommendation_label: str
    explanation: Optional[str] = None

class PanelZoneCreate(PanelZoneBase):
    pass

class PanelZone(PanelZoneBase):
    id: int
    inspection_id: int

    model_config = ConfigDict(from_attributes=True)

class AnalysisSummary(BaseModel):
    total_zones: int
    normal_count: int
    cleaning_priority_count: int
    inspection_required_count: int
    wait_for_rain_count: int
    repair_review_count: int
    average_priority_score: float

class AnalysisResponse(BaseModel):
    inspection_id: int
    status: str
    summary: AnalysisSummary
    top_actions: List[MaintenanceAction]
